### PR TITLE
feat(flow-builder): add stablecoin bridge rails to cross-currency flows

### PR DESCRIPTION
**Pull request by patcapulong**
## Summary

- Cross-currency flows in the flow builder hardcoded the inter-switch leg to BTC over Lightning ("Convert USD to BTC → Send BTC via Lightning → Real-time via Lightning Network"), underselling the orchestration-across-BTC-and-stables pitch (raised by Davi).
- Adds a **"Bridge via"** selector in the sidebar — BTC via Lightning (default), USDC via Solana, USDB via Spark — shown only for two-switch (cross-fiat-region) flows. Picking a rail re-renders the bridge leg: convert/send action cards, the switch-to-switch connector label, and the stablecoin (purple) color treatment.
- **Visualization only.** The quote API has no settlement-rail field (the only rail an integrator can pin is the delivery `paymentRail` on a quote destination), so the generated curl/JSON steps are unchanged — which is itself the pitch: same API calls, Grid handles the routing. A footnote and the bridge-connector tooltip state this explicitly.

## Spec grounding

- USDC on Solana is the spec's documented inter-VASP settlement-leg example (`ReconciliationInstructions.transactionHash`).
- Spark settlement legs carry USDB (`PaymentSparkWalletInfo.assetType`).
- **USDT is deliberately excluded** — it only appears in the API as an external Tron delivery network, with no evidence it's used for inter-switch settlement. Easy to add if confirmed.

Also updates the Grid Switch hover tooltip to mention stablecoins alongside Bitcoin and local rails.

## Test plan

- [x] USD → BRL renders all three bridge rails correctly (cards, connector, colors)
- [x] Single-switch flows (e.g. USD → USDC) unchanged, selector hidden
- [x] Edge cases: BTC source with USDC bridge ("Convert BTC to USDC"), stable source matching the bridge asset ("Accept USDC deposit")
- [x] Generated API steps identical across rail selections
- [x] Visual check on grid-flow-builder.vercel.app preview deploy (light + dark)

Made with [Cursor](https://cursor.com)

### CODE REVIEW

File: components/grid-visualizer/src/lib/flow-path.ts [hunk 1]
```diff
@@ -107,20 +131,17 @@ export function buildFlowPath(
...
+  // over the selected settlement rail — BTC/Lightning by default)
+  const twoSwitches = needsTwoSwitches(source, destination, sourceRegion, destRegion);
 
-  if (needsTwoSwitches) {
+  if (twoSwitches) {
     // Source-side switch
-    const sourceConvertAction: ActionCard = source.code === 'BTC'
-      ? { text: 'Accept BTC deposit', color: 'btc' }
+    const sourceConvertAction: ActionCard = source.code === settle.asset
+      ? { text: `Accept ${settle.asset} deposit`, color: settleColor }
```
[the remark this answers is not in the document]
Comment (patcapulong) [reply components/grid-visualizer/src/lib/flow-path.ts]: Good catch — fixed in 5734de47. Internal sources holding the bridge asset no longer show "Accept X deposit" (the switch just shows the send card; the endpoint card and connector already describe the balance debit), and internal destinations now say "Credit X balance" instead of "Deliver X to wallet", matching how internal accounts render elsewhere. External flows are unchanged.

@greptile-apps review

**Review by carsonp6 — APPROVED** [review]
